source/fuzz/transformation_wrap_vector_synonym.cpp

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) 2021 Shiyu Liu
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "source/fuzz/transformation_wrap_vector_synonym.h"
+#include <algorithm>
+#include "source/fuzz/data_descriptor.h"
+#include "source/fuzz/fuzzer_util.h"
+#include "source/fuzz/instruction_descriptor.h"
+#include "source/opt/instruction.h"
+
+namespace spvtools {
+namespace fuzz {
+
+TransformationWrapVectorSynonym::TransformationWrapVectorSynonym(
+    protobufs::TransformationWrapVectorSynonym message)
+    : message_(std::move(message)) {}
+
+TransformationWrapVectorSynonym::TransformationWrapVectorSynonym(
+    uint32_t instruction_id, uint32_t result_id1, uint32_t result_id2,
+    uint32_t vec_id, uint32_t vec_type_id, uint32_t pos) {
+  message_.set_instruction_id(instruction_id);
+  message_.set_result_id1(result_id1);
+  message_.set_result_id2(result_id2);
+  message_.set_vec_id(vec_id);
+  message_.set_vec_type_id(vec_type_id);
+  message_.set_scalar_position(pos);
+}
+
+bool TransformationWrapVectorSynonym::IsApplicable(
+    opt::IRContext* ir_context, const TransformationContext& /*unused*/) const {
+  auto instruction =
+      ir_context->get_def_use_mgr()->GetDef(message_.instruction_id());
+  auto vector_type =
+      ir_context->get_type_mgr()->GetType(message_.vec_type_id());
+  auto vec1 = ir_context->get_def_use_mgr()->GetDef(message_.result_id1());
+  auto vec2 = ir_context->get_def_use_mgr()->GetDef(message_.result_id2());
+
+  // |instruction_id| must refer to an existing instruction.
+  if (instruction == nullptr) {
+    return false;
+  }
+
+  std::unordered_set<SpvOp_> valid_opcodes{SpvOpIAdd, SpvOpISub, SpvOpIMul,
```

Comment:
SpvOp_ -> SpvOp - better to use the cleaner name of the typedef.

Reply:
Done.